### PR TITLE
Improve the Contributing to Rails Guide [skip ci]

### PR DESCRIPTION
**Pull request by jonatack**
This PR improves the "4.12 Commit Your Changes" section
of the Rails Guide "Contributing to Ruby on Rails".

Changes:
- Fix formatting of the commit message example
- Improve readability and clarity
- Fix grammar
- Wrap lines

Before:
before

After:
after

### CODE REVIEW

File: guides/source/contributing_to_ruby_on_rails.md [hunk 1]
```diff
@@ -399,21 +399,27 @@ When you're happy with the code on your computer, you need to commit the changes
 $ git commit -a
 ```
 
-At this point, your editor should be fired up and you can write a message for this commit. Well formatted and descriptive commit messages are extremely helpful for the others, especially when figuring out why given change was made, so please take the time to write it.
+This should open a buffer in your editor for writing a commit message (when you
```
Comment (vijaydev) [inline guides/source/contributing_to_ruby_on_rails.md]: 'buffer' sounds very vim/emacs like.. I prefer the existing version.

Comment (jonatack) [inline guides/source/contributing_to_ruby_on_rails.md]: Thanks for reviewing! Would 'file' or 'window' be better?

The change in this paragraph addressed 2 things:

1) the current version is ambiguous for someone who might be doing this for the first time: whether the editor should already be fired up or if it is the git command that fires it up, and what the user should do to continue.

2) The importance of writing a good commit message is a different topic than using the editor and deserves its own paragraph.

Comment (carlosantoniodasilva) [inline guides/source/contributing_to_ruby_on_rails.md]: Maybe `This should open your editor` should be enough?

Comment (jonatack) [inline guides/source/contributing_to_ruby_on_rails.md]: "This should fire up your editor to write a commit message; when you have finished, save and close to continue."

Comment (jonatack) [inline guides/source/contributing_to_ruby_on_rails.md]: OK, fire up or open? (to me, one fires up an editor and opens a buffer or window).

"This should fire up your editor to write a commit message; when you have finished, save and close to continue."

Comment (carlosantoniodasilva) [inline guides/source/contributing_to_ruby_on_rails.md]: I think buffer or window are details that are not super important in the context of the guide, stating that the editor will be opened/fired up seems to be enough to understand it. I don't have strong opinion on `open vs fire up`.

**Comment by jonatack** [comment]
Thanks for the feedback! Updated and squashed.

File: guides/source/contributing_to_ruby_on_rails.md (hunk 1 above)
Comment (jonatack) [inline guides/source/contributing_to_ruby_on_rails.md]: Thanks @vijaydev and @carlosantoniodasilva.

File: guides/source/contributing_to_ruby_on_rails.md [hunk 2]
```diff
@@ -399,21 +399,27 @@ When you're happy with the code on your computer, you need to commit the changes
...
-characters. Try to be as descriptive as you can; even if you think that the
-commit content is obvious, it may not be obvious to others. Add any description
-that is already present in relevant issues - it should not be necessary to visit
-a webpage to check the history.
+More detailed description, if necessary. It should be wrapped to
+72 characters. Try to be as descriptive as you can. Even if you
+think that the commit content is obvious, it may not be obvious
+to others. Add any description that is already present in the
+relevant issues; it should not be necessary to visit a webpage
+to check the history.
```
Comment (matthewd) [inline guides/source/contributing_to_ruby_on_rails.md]: Shouldn't this paragraph be wrapped to 72 characters? :confused:

Comment (jonatack) [inline guides/source/contributing_to_ruby_on_rails.md]: @matthewd I wrapped it as wide as possible while still rendering correctly (see the first attached image).

**Comment by jonatack** [comment]
Rendered section after update:

updated-pr

File: guides/source/contributing_to_ruby_on_rails.md (hunk 2 above)
Comment (jonatack) [inline guides/source/contributing_to_ruby_on_rails.md]: The line width for git commits and the line width for Markdown `code` sections in the Rails Guides do not appear to be related.
